plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -0,0 +1,570 @@
+import java.net.URL;
+
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.HttpHost;
+import org.apache.http.client.utils.URIUtils;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.HttpStatus;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.WildcardFileFilter;
+import org.apache.commons.io.IOUtils;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+import io.jenkins.tools.pluginmanager.config.Config;
+
+
+public class PluginManager {
+    private List<Plugin> plugins;
+    private List<Plugin> failedPlugins;
+    private File refDir;
+
+    private String JENKINS_UC_LATEST = "";
+    private final String JENKINS_UC_EXPERIMENTAL = "https://updates.jenkins.io/experimental";
+    private final String JENKINS_INCREMENTALS_REPO_MIRROR = "https://repo.jenkins-ci.org/incrementals";
+    private final String JENKINS_UC = "https://updates.jenkins.io";
+    private final String JENKINS_UC_DOWNLOAD = JENKINS_UC + "/download";
+    private final String JENKINS_UC_JSON = "https://updates.jenkins.io/update-center.json";
+    private final String SEPARATOR = File.separator;
+
+    //private final String JENKINS_WAR = "/usr/share/jenkins/jenkins.war";
+    public static final String JENKINS_WAR = "/Users/natashastopa/Jenkins/Jenkins/test2/mywar-1.0-SNAPSHOT.war";
+
+    private String jenkinsVersion;
+
+    private File jenkinsWarFile;
+    private Map<String, String> installedPluginVersions;
+    private Map<String, String> bundledPluginVersions;
+    private List<SecurityWarning> allSecurityWarnings;
+    Config cfg;
+
+
+    public PluginManager(Config cfg) {
+        plugins = new ArrayList<>();
+        failedPlugins = new ArrayList();
+        jenkinsWarFile = new File(JENKINS_WAR);
+        installedPluginVersions = new HashMap<>();
+        bundledPluginVersions = new HashMap<>();
+        refDir = cfg.getPluginDir();
+        this.cfg = cfg;
+        allSecurityWarnings = new ArrayList<>();
+    }
+
+
+    public void start() {
+        if (!refDir.mkdir()) {
+            System.out.println("Unable to create plugin directory");
+        }
+
+        jenkinsVersion = getJenkinsVersion();
+        checkVersionSpecificUpdateCenter();
+
+        String url;
+
+        getSecurityWarnings();
+
+        if (cfg.hasShowAllWarnings()) {
+            for (int i = 0; i < allSecurityWarnings.size(); i++) {
+                SecurityWarning securityWarning = allSecurityWarnings.get(i);
+                System.out.println(securityWarning.getName() + " " + securityWarning.getMessage());
+            }
+        }
+
+        System.out.println("Reading in plugins...");
+        try {
```

Comment:
I was playing around with using the command line, and I hit this since I didn't run with a plugins.txt file.  And you specifically called it out in a meeting (I think)  This should be pulled into the CLI class because library users won't necessarily be using a plugin text file.

I'm ok with pulling that conversion out into another ticket.

Reply:
I think it will be pretty easy to move the current parsing over to the CLI, which will make it easier to combine all the plugins the user wants to install from both the command line input and the plugins.txt into a list that can be added to the config file. I think doing this will make it a little more demo ready so I will add it to this PR in my next commit along with a fix to the previous comment of not doing anything with the plugins passed directly into the CLI.

I think I already create a ticket to Jira for this, but I think there's still a lot of work to be done in terms of making the parsing more robust and being able to parse different types of input files, which we can do at a later point.

Reply:
Moved basic parsing in 7ca3d5d987f6c60617f5f95994574849fb17f170.